### PR TITLE
Add Roles and Users endpoints integration tests

### PR DESCRIPTION
**Pull request by joshcanhelp**
### Changes

- Add integration tests (recorded with VCR) for new Roles and Users endpoints.
- Rename `unlink_users_account` to `unlink_user_account`, added alias and test
- Add `delete_with_body` method to send a body rather than params with a `DELETE` request

**Note for reviewers:**

- New VCR recordings were add in `spec/fixtures/vcr_cassettes/Auth0_Api_V2_Users/` and `spec/fixtures/vcr_cassettes/Auth0_Api_V2_Roles/`. Sensitive information (API token) is automatically filtered out. 
- The other YML recording files have either a changed format, changed date, or removed dummy API token (the one that was there is not valid but makes more sense for them all to have `API_TOKEN`). 
- In general, YML files can be ignored. Without those, review is ~360 changed lines total. 

### References

- User methods added in #169 (adjust naming in #173 and #175)
- Role methods added in #172 (adjust naming in #173)

### Testing

* [ ] This change adds unit test coverage (added in original PR)
* [x] This change adds integration test coverage
* [x] This change has been tested on the latest version of Ruby (2.6.0)

### CODE REVIEW

File: lib/auth0/api/v2/users.rb [hunk 1]
```diff
@@ -220,19 +221,19 @@ def remove_user_roles(user_id, roles)
           raise Auth0::MissingUserId, 'Must supply a valid user_id' if user_id.to_s.empty?
           validate_strings_array roles
           path = "#{users_path}/#{user_id}/roles"
-          delete(path, { roles: roles })
+          delete_with_body path, roles: roles
```
Comment (lbalmaceda) [inline lib/auth0/api/v2/users.rb]: I see inconsistencies on the syntax when calling a function with parameters.

`delete(path, {param: value}` vs `delete path, param:value`. 

Should we add a linter as a precommit for this repo as well or is this change on purpose?

File: lib/auth0/mixins/httpproxy.rb [hunk 2]
```diff
@@ -20,6 +20,8 @@ module HTTPProxy
                      call(:get, url(safe_path), timeout, get_headers)
                    elsif method == :delete
                      call(:delete, url(safe_path), timeout, add_headers({params: body}))
```
Comment (lbalmaceda) [inline lib/auth0/mixins/httpproxy.rb]: what is this `params:body` header for? I guess you meant to send a query instead?

Comment (joshcanhelp) [reply lib/auth0/mixins/httpproxy.rb]: Oddly, sending it as `params:` converts it to a query ... one of the many quirks of this HTTP library

File: lib/auth0/api/v2/users.rb (hunk 1 above)
Comment (joshcanhelp) [reply lib/auth0/api/v2/users.rb]: There is a linter that I run, `rubocop`, but it doesn't pick this up
